plugins/catalog/src/hooks/useEntities.ts

```diff
@@ -40,6 +40,88 @@ type UseEntities = {
   selectTypeFilter: (id: string) => void;
 };
 
+type EntityFilterGroupOutput<T> = {
+  selectItems: (items: string[]) => void;
+  filteredItems: T[];
+  states: OutputState;
+};
+
+type OutputState = { [key: string]: { isSelected: boolean; count: number } };
+
+type FilterDefinition<T> = {
+  [key: string]: {
+    isSelected: boolean;
+    filterFunction: (entity: T) => boolean;
+  };
+};
+
+export const useEntityFilterGroup = <T>(
+  entities: T[],
```

Comment:
this one won't be able to take the entities as input; it's rather the framework (the context) that will supply entities in the output

Reply:
I still don't get the need for context. Let's discuss.